compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -5154,8 +5152,35 @@ abstract class GenJSCode extends plugins.PluginComponent
             js.OptimizerHints.empty, None)
       }
 
+      // Compute the set of method symbols that we need to implement
+      val sams = {
+        val samsBuilder = List.newBuilder[Symbol]
+        val seenEncodedNames = mutable.Set.empty[String]
+
+        /* scala/bug#10512: any methods which `samInfo.sam` overrides need
+         * bridges made for them.
+         * On Scala < 2.12.5, `synthCls` is polyfilled to `NoSymbol` and hence
+         * `samBridges` will always be empty. This causes our compiler to be
+         * bug-compatible on these versions.
+         */
+        val synthCls = samInfo.synthCls
+        val samBridges = if (synthCls == NoSymbol) {
+          Nil
+        } else {
+          import scala.reflect.internal.Flags.BRIDGE
+          synthCls.info.findMembers(excludedFlags = 0L, requiredFlags = BRIDGE).toList
+        }
+
+        for (sam <- samInfo.sam :: samBridges) {
+          if (seenEncodedNames.add(encodeMethodSym(sam).name))
```

Comment:
Why do we need this?

Reply:
Because the JVM backend does something similar (they call `.disctinct` [here](https://github.com/scala/scala/pull/6087/files#diff-2e0fd5042aa83407a80a7557cc2a666bR1375)). I guess this would happen if a SAM type extends several interfaces, each defining the same method, which would then collapse as the same bridge. The most reliable way for us to detect duplicates at this point is to test uniqueness of the `encodedName`.

Reply:
Hmmm... OK. Not very nice, but fair enough.